### DISCUSSION TITLE
Query script errors not highlighted, only position count provided.

### DISCUSSION THREAD

**Discussion by coleman-rik** [dbeaver/dbeaver#11907] [Q&A]
#### System information: 
Windows 10 x64 Enterprise
- DBeaver 7.2.4.202011011603

#### Connection specification:
PostgreSQL 11
org.postgresql:postgresql:RELEASE [42.2.18]

- Do you use tunnels or proxies (SSH, SOCKS, etc)? 
No

#### Describe the problem you're observing:
When running queries in DBeaver, whenever there's an error all DBeaver provides is a general description of the error and a Position: xxx.  It doesn't provide a line number nor any indication of *which* query has an error at Position: xxx.

Unfortunately when running large queries (the one I am working on at the moment is over 1,000 lines comprised of more than 20 individual queries) this doesn't help identify where DBeaver thinks the error may be.  This is compounded by the fact that the query editor in DBeaver doesn't have any indication in the status bar (or anywhere else I can find) at least telling you what position the cursor is currently *at*.

Running the same query against the same database in pgAdmin4 returns the same error message as expected.  At least there the error message provides the line number, the character number in the script as a whole and scrolls the query editor to the problematic line.  It even highlights it in pink.  So I find myself writing and running queries in DBeaver, but having to rerun those same queries in pgAdmin4 whenever there's an error to get any idea where or what the error might be.

#### Steps to reproduce, if exist:
Create a query with an error.
Try to run that query.
Try and figure out where Position: xxx is that DBeaver thinks the error lies.

For real fun; 
Create 10 or 20 or more queries in the same query editor (say 1200 lines) with an error.
User the 'Run Script' functionality to run all of them.
Marvel at how after running for a couple of minutes DBeaver claims that there is an error and gives you some seemingly random position like 'Position: 336'

**Comment by kseniiaguzeeva** [comment]
Thank you for the report. Highlighting is not implemented yet. We are going to do it.

**Comment by ShadelessFox** [comment]
Hello, @coleman-rik,
The error message you see is reported by PostgreSQL itself. In case of PostgreSQL, your cursor is moved to the position which specified in the error message. It may vary between different databases, e.g. in SQL Server only error line is reported, so entire line will be selected.
Also you can find current cursor location in the form of `line:column:offset` under the SQL editor.
Let me know if your problem was solved.

**Comment by uslss** [comment]
there is no update for more than a month.
If the issue is still actual for you - feel free to reopen it

**Comment by uslss** [comment]
#11869 
DBeaver 21.0.1 It now highlights the first error (which is an improvement). The bottom of the window now lists the current cursor position as Line: character along line: character from beginning of statement. Unfortunately only indication it gives of the location of the error is still just a position (which I am assuming is in characters from the beginning of the statement) ex: Position: 162. It would be very helpful if it could provide the location of the error in a format that matches that in the bottom of the window. Additionally, when parsing a statement and there is a syntax error, it would be helpful if it listed all of the errors encountered, not just the first one.

Thanks.

**Comment by ShadelessFox** [comment]
Hello @coleman-rik,

We don't parse the statements by ourselves, the errors are sent by the server itself, as mentioned multiple times before. The same thing is for the error message - the code position is the part of it, we don't modify it.

**Reply by coleman-rik** [reply]
That's unfortunate.  I was hoping that, knowing the text and the arrangement of the query in the query window dBeaver could calculate which **line : position in line : position** the error would end up in to match the format chosen for the bottom since dBeaver can calculate this position of the cursor.

Well, keep up the good work.
